pkgs/development/compilers/fbc/default.nix

```diff
@@ -0,0 +1,152 @@
+{ stdenv
+, buildPackages
+, lib
+, fetchzip
+, fetchpatch
+, gpm
+, libffi
+, libGL
+, libX11
+, libXext
+, libXpm
+, libXrandr
+, ncurses
+}:
+
+stdenv.mkDerivation rec {
+  pname = "fbc";
+  version = "1.09.0";
+
+  src = fetchzip {
+    # Bootstrap tarball has sources pretranslated from FreeBASIC to C
+    url = "https://github.com/freebasic/fbc/releases/download/${version}/FreeBASIC-${version}-source-bootstrap.tar.xz";
+    sha256 = "1q1gxp5kjz4vkcs9jl0x01v8qm1q2j789lgvxvikzd591ay0xini";
+  };
+
+  patches = [
+    # Fixes fbc_tests.udt_wstring_.midstmt ascii getting stuck due to stack corruption
+    (fetchpatch {
+      url = "https://github.com/freebasic/fbc/commit/42f4f6dfdaafdd5302a647152f16cda78e4ec904.patch";
+      excludes = [ "changelog.txt" ];
+      sha256 = "sha256-Bn+mnTIkM2/uM2k/b9+Up4HJ7SJWwfD3bWLJsSycFRE=";
+    })
+  ];
+
+  postPatch = ''
+    patchShebangs tests/warnings/test.sh
+
+    # Some tests lack proper dependency on libstdc++
+    for missingStdcpp in tests/cpp/{class,call2}-fbc.bas; do
+      sed -i -e "/'"' TEST_MODE : /a #inclib "stdc++"' $missingStdcpp
+    done
+
+    # Help compiler find libstdc++ with gcc backend
+    sed -i -e '/fbcAddLibPathFor( "libgcc.a" )/a fbcAddLibPathFor( "libstdc++.so" )' src/compiler/fbc.bas
+
+    # r13y of date/time macros
+    substituteInPlace src/compiler/symb-define.bas \
+      --replace 'function = date' "return \"$(date -ud "@$SOURCE_DATE_EPOCH" '+%m-%d-%4Y')\"" \
+      --replace 'function = format( now( ), "yyyy-mm-dd" )' "return \"$(date -ud "@$SOURCE_DATE_EPOCH" '+%F')\"" \
+      --replace 'function = time' "return \"$(date -ud "@$SOURCE_DATE_EPOCH" '+%T')\""
+  '';
+
+  dontConfigure = true;
+
+  depsBuildBuild = [
+    buildPackages.stdenv.cc
+    buildPackages.ncurses
+    buildPackages.libffi
+  ];
+
+  buildInputs = [
+    ncurses
+    libffi
+  ] ++ lib.optionals stdenv.hostPlatform.isLinux [
+    gpm
+    libGL
+    libX11
+    libXext
+    libXpm
+    libXrandr
+  ];
+
+  enableParallelBuilding = true;
+
+  # Some tests seem to fail with too much (~64?) parallelism
```

Comment:
Could we set an upper bound?

Reply:
We could but it'll require some testing on ofborg to find a good value, and making sense of the check output when all tests are run in parallel will be impossible… I'll see what I can do.

Reply:
Okay, seems to work.